### ISSUE TITLE
min-vid uses eval(), causing a violation of CSP

### ISSUE THREAD

**Issue by andreastt** [meandavejustice/min-vid#582]
When a site has Content Security Policy (CSP) enabled which restricts uses of `eval()`, a warning is emitted to the console:

> Content Security Policy: The page’s settings blocked the loading of a resource at self (“script-src https://sny.no”). Source: call to eval() or related function blocked by CSP.

The CSP rule in place for [my website](https://sny.no) is:

    Content-Security-Policy: default-src 'self' 'unsafe-inline'; script-src 'self'; object-src 'none'; frame-ancestors 'none'

The relevant part of this policy is the `script-src 'self'`. If you add `'unsafe-eval'`, the error goes away.

**Comment by moggers87** [comment]
Getting a similar error, but for style:

> Content Security Policy: The page's settings blocked the loading of a resource at self ("style-src https://192.168.2.135"). Source: 
.minvid__overlay__container {
    align....

CSP header from the server was:

```
Content-Security-Policy: script-src 'self'; style-src 'self'; default-src 'none'; img-src 'self'; connect-src 'self'; font-src 'self'; report-uri /_csp_report/
```

**Comment by freddyb** [comment]
I can reproduce on a website of mine.

Steps to Reproduce:
1) Go to <https://locks.on.web.security.plumbing/>
2) Open DevTools
3) See CSP violation (like above, `.minvid__overlay__container ....`)

**Comment by meandavejustice** [comment]
@freddyb This is because we inject the stylesheet into pages with youtube embeds to show the overlay icon. https://github.com/meandavejustice/min-vid/blob/master/webextension/content-scripts/icon-overlay.js#L494

Not sure how we can get around this while we are an extension :/

**Comment by moggers87** [comment]
Would it be possible to only inject the CSS when a valid video is found?

**Comment by freddyb** [comment]
Yes, it should.

**Comment by meandavejustice** [comment]
@moggers87 yeah, we should be doing that, I'll write a patch for that today to be included in the next release
